src/utils/work/seekWorkResponse.js

```diff
@@ -176,8 +176,67 @@ const dispatchWorkRequest = async (
     },
   );
 
-  // TODO switch to using normal WorkRequest for v2 requests
   return await Promise.race([timeoutPromise, responsePromise]);
 };
 
-export { dispatchWorkRequest, seekFromS3 };
+const seekWorkerResultsOrDispatchWork = async (
+  experimentId,
+  body,
+  timeout,
+  requestProps,
+  ETagProps,
+  onETagGenerated,
+  dispatch,
+) => {
+  const url = `/v2/workResults/${experimentId}`;
+
+  const workResults = await fetchAPI(url, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(ETagProps),
+  });
+
+  const { ETag } = workResults;
+  let { signedUrl } = workResults;
+
+  onETagGenerated(ETag);
+
+  // First, let's try to fetch this information from the local cache.
+  let data = await cache.get(ETag);
+
+  if (data) {
+    return { ETag, data };
+  }
+
+  // If a signed url is returned, we know that results already exist
+  if (signedUrl) {
+    data = await downloadFromS3(signedUrl);
+    return { ETag, data };
+  }
+
+  // Otherwise subscribe to the worker socket using the Etag
+  try {
```

Comment:
remove this try catch as it only makes the code harder to read and doesn't handle the error (just printing a console log)